test/cluster/rack_revision_test.go

```diff
@@ -0,0 +1,486 @@
+
+	racks := []asdbv1.Rack{
+		{ID: 1, Revision: revision},
+		{ID: 2, Revision: revision},
+	}
+
+	rackConf := asdbv1.RackConfig{
+		Racks: racks,
+	}
+
+	aeroCluster.Spec.RackConfig = rackConf
+
+	return aeroCluster
+}
+
+func validateRackRevisionStatefulSets(
+	k8sClient client.Client, ctx goctx.Context, clusterNamespacedName types.NamespacedName,
+	revision string, expectedCount int,
+) error {
+	stsList := &appsv1.StatefulSetList{}
+	labelSelector := labels.SelectorFromSet(utils.LabelsForAerospikeCluster(clusterNamespacedName.Name))
+	listOps := &client.ListOptions{
+		Namespace:     clusterNamespacedName.Namespace,
+		LabelSelector: labelSelector,
+	}
+
+	if err := k8sClient.List(ctx, stsList, listOps); err != nil {
+		return err
+	}
+
+	var revisionCount int
+
+	for idx := range stsList.Items {
+		if rackRevision, exists := stsList.Items[idx].Labels[asdbv1.AerospikeRackRevisionLabel]; exists &&
+			rackRevision == revision {
+			revisionCount++
+		}
+	}
+
+	if revisionCount != expectedCount {
+		return fmt.Errorf("expected %d StatefulSets with revision %s, found %d", expectedCount, revision, revisionCount)
+	}
+
+	return nil
+}
+
+func checkBothRevisionsExist(
+	k8sClient client.Client, ctx goctx.Context, clusterNamespacedName types.NamespacedName, oldRevision,
+	newRevision string,
+) bool {
+	stsList := &appsv1.StatefulSetList{}
+	labelSelector := labels.SelectorFromSet(utils.LabelsForAerospikeCluster(clusterNamespacedName.Name))
+	listOps := &client.ListOptions{
+		Namespace:     clusterNamespacedName.Namespace,
+		LabelSelector: labelSelector,
+	}
+
+	if err := k8sClient.List(ctx, stsList, listOps); err != nil {
+		return false
+	}
+
+	var oldExists, newExists bool
+
+	for idx := range stsList.Items {
+		if rackRevision, exists := stsList.Items[idx].Labels[asdbv1.AerospikeRackRevisionLabel]; exists {
+			if rackRevision == oldRevision {
+				oldExists = true
+			}
+
+			if rackRevision == newRevision && stsList.Items[idx].Status.ReadyReplicas > 1 {
+				newExists = true
+			}
+		}
+	}
+
+	return oldExists && newExists
+}
+
+func validateRackRevisionCleanup(
+	k8sClient client.Client, ctx goctx.Context, aeroCluster *asdbv1.AerospikeCluster,
+	rackIDs []int, cleanedRevision string,
+) error {
+	for _, rackID := range rackIDs {
+		oldResourceNsNm := GetNamespacedNameForSTS(aeroCluster, utils.GetRackIdentifier(rackID, cleanedRevision))
+
+		oldSts := &appsv1.StatefulSet{}
+		nsNm := types.NamespacedName{Namespace: oldResourceNsNm.Namespace, Name: oldResourceNsNm.Name}
+
+		if err := k8sClient.Get(ctx, nsNm, oldSts); err == nil {
+			return fmt.Errorf("StatefulSet with revision %s still exists after cleanup: %s", cleanedRevision, nsNm.Name)
+		}
+
+		cm := &corev1.ConfigMap{}
+
+		if err := k8sClient.Get(ctx, nsNm, cm); err == nil {
+			return fmt.Errorf("ConfigMap with revision %s still exists after cleanup: %s", cleanedRevision, nsNm.Name)
+		}
+	}
+
+	return nil
+}
+
+func deleteOldStatefulSet(
```

Comment:
you can rename it more generic, as you are passing which revision you want to delete. No need to mention Old in name.